ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -168,11 +181,20 @@ a database ( directory )
           functionsTracker.debugLog("functions");
           break;
         }
+        case Constraint: {
+          LoadConstraint loadConstraint =
+              new LoadConstraint(context, (ConstraintEvent) next, work.dbNameToLoadIn, dbTracker);
+          TaskTracker constraintTracker = loadConstraint.tasks();
+          scope.rootTasks.addAll(constraintTracker.tasks());
```

Comment:
constraint task will start only when all other tasks are processed.

Reply:
How is it guaranteed as all tasks such as table load, functions load, constraints load are added as root tasks which have same hierarchy right?

Reply:
Constraint will be in separate batch, it is bounded by addAnotherLoadTask. My understanding is it will only start run after all previous batches finishes, is that right?

Reply:
oh yes. Noticed it now. The while loop conditions are quiet misleading. Can we explicitly mark the hierarchy with parenthesis such as (iterator.hasNext() || (loadingConstraint && constraintIterator.hasNext()))?

Reply:
Yeah, you are right.